src/App.js

```diff
@@ -1,16 +1,85 @@
-import React from 'react';
+import React, {useState} from 'react';
 import './App.css';
 import chatMessages from './data/messages.json';
+import ChatLog from './components/ChatLog';
+import ColorChoice from './components/ColorChoice';
 
 const App = () => {
+  const[entriesData, setEntriesData] = useState(chatMessages);
+
+  const likeEntry = (id) => {
+    setEntriesData((prev)=>{
+      return prev.map((entry)=>{
+        if(id === entry.id){
+          return{
+            ...entry,
+            liked: !entry.liked
+          };}
+        else{
+          return entry;
+        }
+        });
+    });
+  };
+
+  const totalLikes= entriesData.reduce((total, entry)=>{
+    if(entry.liked === true){
+      total +=1;
+    }
+    return total;
+  }, 0);
+
+  const [senderColors, setSenderColors] = useState({
+    [chatMessages[0].sender]: 'black',
+    [chatMessages[1].sender]: 'black'
+  });
+
+  const handleSenderColor = (sender, color) => {
+    setSenderColors((prevColors) => ({
+      ...prevColors,
+      [sender]: color
+    }));
+    
+    //I updated the data to set the text color dinamically
+    const updatedEntriesData = entriesData.map((entry) => {
+      if (entry.sender === sender) {
+        return {
+          ...entry,
+          color: color
+        };
+      } else {
+        return entry;
+      }
+    });
+  
+    setEntriesData(updatedEntriesData);
+  };
+
   return (
     <div id="App">
       <header>
-        <h1>Application title</h1>
+        <h1>
+          Chat between{' '}
+          <span className={senderColors[chatMessages[0].sender]}>{chatMessages[0].sender}</span>{' '}
+          and{' '}
+          <span className={senderColors[chatMessages[1].sender]}>{chatMessages[1].sender}</span>
```

Comment:
👍 Nice dynamic header. But what would happen if there weren't messages in the conversation yet. Or what if one participant starts the conversation by sending mulltiple messages?

These would be points to consider anywhere the code is diresctly accessing indices 0 and 1 of the chat message data.